### PR TITLE
Fix WASI builds with abc enabled

### PR DESCRIPTION
**Pull request by whitequark**
This PR works around #2011 and updates ABC to a more recent commit with WASI support.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -281,13 +281,13 @@ yosys.html: misc/yosys.html
 
 else ifeq ($(CONFIG),wasi)
 ifeq ($(WASI_SDK),)
-CXX = clang++
+CXX = clang
```
Comment (eddiehung) [inline Makefile:284]: Why this change?

Comment (whitequark) [reply Makefile:284]: Because the Yosys build system is [fundamentally broken](https://github.com/YosysHQ/yosys/issues/2011) at the moment. All the other configurations alo have `CXX = <c-compiler-frontend>` in them.

Comment (eddiehung) [reply Makefile:284]: Ah, I see this is the default:
https://github.com/YosysHQ/yosys/blob/ff7a8d0e1e1e6fab3e508bb8c237502ef570132b/Makefile#L185
